packages/nc-gui-v2/utils/projectCreateUtils.ts

```diff
@@ -4,30 +4,15 @@ export interface ProjectCreateForm {
   title: string
   dataSource: {
     client: ClientType
-    connection:
-      | {
-          host: string
-          database: string
-          user: string
-          password: string
-          port: number | string
-          ssl?: Record<string, string>
-        }
-      | {
-          client?: ClientType.SQLITE
-          database: string
-          connection?: {
-            filename?: string
-          }
-          useNullAsDefault?: boolean
-        }
+    connection: Record<string, any>
```

Comment:
Now all these properties are basically "unknown" to us and I'd have to guess which properties belong to one another.
Instead of removing all this - which we could still use to improve the actual typing that can differentiate better between each one

For example
```ts
enum ClientType {
  NONE = 'none',
  MYSQL = 'mysql2',
  MSSQL = 'mssql',
  PG = 'pg',
  SQLITE = 'sqlite3',
  VITESS = 'vitess',
}

interface DefaultConnection {
	client?: ClientType.NONE,
	host: string
        database: string
        user: string
        password: string
        port: number | string
        ssl?: Record<certTypes, string> | 'true'
}

interface SQLiteConnection {
	client?: ClientType.SQLITE
        database: string
        connection?: {
           filename?: string
        }
        useNullAsDefault?: boolean
}

type Connections = DefaultConnection | SQLiteConnection 


...

	dataSource: {
		connection: Connections
		...
	}
```

Now if each Connection shares the fact that they have the `client` property, we could always differentiate between them by asking what client type is currently assigned.

```ts
if (formState.dataSource.connection.client === ClientType.None) { ... }
else if (formState.dataSource.connection.client === ClientType.SQLITE) { ... }
```

If it's about the `onSSLModeChange` issue - if `.ssl` is supposed to be writable for *any* type of Connection, it should also be included in each type that is possible - otherwise it's incorrect and TypeScript rightfully complains about it 😄 

Otherwise with the suggested changes this would work
```ts
const onSSLModeChange: SelectHandler = (mode: sslUsage) => {
  if (formState.dataSource.connection.client === ClientType.NONE) {
    switch (mode) {
      case sslUsage.No:
        delete formState.dataSource.connection.ssl
        break
      case sslUsage.Allowed:
        formState.dataSource.connection.ssl = 'true'
        break
      default:
        formState.dataSource.connection.ssl = {
          ca: '',
          cert: '',
          key: '',
        }
        break
    }
  }
}
```

Reply:
You could also, if necessary, type cast `formState.dataSource.connection` to `DefaultConnection` - at least we have some typesafety then 😅